Home_Work13/HW13.cpp

```diff
@@ -0,0 +1,442 @@
+#define _CRT_SECURE_NO_WARNINGS
+#include <iostream>
+#include <fstream>
+#include <string>
+#include <algorithm>
+#include <ctime>
+
+enum class readParameters //defines which string to read from save file
+{
+	Date,
+	Word,
+	Status
+};
+
+void printMainMenu();
+void printGameStartedMenu(int attempts);
+std::string mainModeErrors(std::string mainModeChoise);
+std::string gameModeErrors(std::string inputString, int wordLength);
+std::string intermidiateWordResult(std::string inputString, std::string guessWord, const int wordLength);
+std::string generateWord(const std::string WordsFile, const int WordsCount, const int wordLengt);
+void unmatchedLettersList(std::string inputString, std::string guessWord, const int wordLength, std::string& usedUnmatchLetters);
+bool checkWinCondition(std::string guessWord, std::string resultAfterGuess, const int wordLength);
+void printNewRoundMenu(std::string inputString, std::string resultAfterGuess, std::string usedUnmatchLetters, int currentAttemptsCounter);
+void printWinMenu(std::string guessWord, const int maxAttempts, int currentAttemptsCounter);
+void printlossMenu(std::string guessWord, std::string mainModeChoise);
+void printError(std::string errors);
+std::string readSavedData(std::string saveDataToFile, readParameters parameterToRead);
+std::string getCurrentDate();
+void writeSaveData(std::string date, std::string word, std::string status, const std::string saveDataToFile);
+
+int main()
+{
+	const int usualWordsCount = 212; //number of words for usual mode, changing this requires changing number of words in data base
+	const int wordsOfTheDayCount = 219; //number of words for word of the day mode, changing this requires changing number of words in data base
+	const int wordLength = 5; //word length, changing this requires to update words data base
+	const int maxAttempts = 10; //number of attempts to guess the word
+
+	const std::string usualWordsFile = "UsualWords.txt"; //file with words data base
+	const std::string wordOfTheDayFile = "DayWord.txt"; //file with words data base
+	const std::string saveDataToFile = "SaveData.txt"; //file with save data
+
+	while (true)
+	{
+		std::string today = getCurrentDate(); 
+
+		printMainMenu();
+
+		std::string mainModeChoise = "";
+		std::cin >> mainModeChoise;
+		std::cin.ignore(1000, '\n');
+		std::string mainModeError = mainModeErrors(mainModeChoise); // form the eeror string
+
+
+		std::string guessWord = "";
+		std::string wordOfTheDayGuessStatus = "false"; 
+
+		if (mainModeError != "") //if error string now empty this mean there are error and program will restart
+		{
+			printError(mainModeError);
+			continue;
+		}
+
+		else if (mainModeChoise == "0")  // if parameter is 0 program will ends
```

Comment:
can create another enum class for modeChoice and don't use this magic numbers with comments explaining its meaning